addons/point_of_sale/static/src/app/store/db.js

```diff
@@ -288,6 +289,11 @@ export class PosDB {
             if (product.barcode && product.active) {
                 this.product_by_barcode[product.barcode] = product;
             }
+            if (this.product_by_tmpl_id[product.product_tmpl_id]) {
+                this.product_by_tmpl_id[product.product_tmpl_id].push(product);
+            } else {
+                this.product_by_tmpl_id[product.product_tmpl_id] = [product];
+            }
```

Comment:
to see with JS, but I really don't like this way of "append to the list if the key does not exists" kind.
Personally I would prefer the equivalent:
`(this.product_by_tmpl_id[product.product_tmpl_id] ??= []).push(product)`
source:
https://stackoverflow.com/a/64608915

@pedrambiria what do you think on this :thinking:

Reply:
@lse-odoo  Are we waiting for @pedrambiria  opinion or should I make the change?

Reply:
@lse-odoo your suggestion is better, but I'm not sure if it's working on all of the browsers.